File: 744. Find Smallest Letter Greater Than Target/solution.py
class Solution(object):
  def nextGreatestLetter(self, letters, target):
    """
    :type letters: List[str]
    :type target: str
    :rtype: str
    """
    if target >= letters[-1]:
      return letters[0]
    
    for letter in letters:
      if target < letter:
        return letter

File: 744. Find Smallest Letter Greater Than Target/solution.java
class Solution {
  public char nextGreatestLetter(char[] letters, char target) {
    if (target >= letters[letters.length - 1]){
      return letters[0];
    }

    for (int i = 0; i < letters.length; i++){
      char letter = letters[i];
      if (target < letter){
        return letter;
      }
    }
    return letters[0];
  }
}